File: baseclasslib/src/main/java/com/laboratory/baseclasslib/base/BaseFragment.java
package com.laboratory.baseclasslib.base;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

abstract class BaseFragment extends Fragment {

    protected abstract void initView();

    protected abstract void initData();

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        initView();
        initData();
    }
}

File: baseclasslib/src/main/java/com/laboratory/baseclasslib/base/BaseFragment.kt
package com.laboratory.baseclasslib.base

import android.os.Bundle
import androidx.fragment.app.Fragment

abstract class BaseFragment : Fragment() {

    protected abstract fun initView()

    protected abstract fun initData()

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        initView()
        initData()
    }
}